### PR TITLE
Track `@types/*` packages

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Create a regularly updating issue (#6965) that tracks all `major.minor` versions for packages with a matching `@types/*`.

## Why?

Mismatches in @types packages significantly reduce any type safety gained by using TypeScript.

## Screenshots

image

### CODE REVIEW

File: scripts/deno/types-dependencies.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * We only look at major.minor versions as per DefinitelyTyped docs:
+ * https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library
+ */
+const majorMinorVersions = Object.entries(pkg.dependencies).map(
+	([dependency, version]) =>
+		[dependency, version.match(/(\d+\.\d+)/)?.at(0)] as const,
+);
+
+// console.log(majorMinorVersions);
```
Comment (jamesgorrie) [inline scripts/deno/types-dependencies.ts:15]: 👀 ?

**Review by jamesgorrie — COMMENTED** [review]
How do we see this working in terms of telling us when things are wrong? The comment is really useful, but what could be really _really_ useful is having alerting for when we regress, or even a weekly rundown?

**Comment by joecowton1** [comment]
> How do we see this working in terms of telling us when things are wrong? The comment is really useful, but what could be really _really_ useful is having alerting for when we regress, or even a weekly rundown?

Agree, also really needs an investment of time to resolve some of the wider issues at play here - visibility is great but it doesn't fix anything.

I think when everyone is ready to upgrade to React 18 there will scope to eliminate at least a few of the mismatches.

**Comment by mxdvl** [comment]
> How do we see this working in terms of telling us when things are wrong? The comment is really useful, but what could be really really useful is having alerting for when we regress, or even a weekly rundown?

> Agree, also really needs an investment of time to resolve some of the wider issues at play here - visibility is great but it doesn't fix anything.

I’ve now merged the tilde range CI action from #6971, @joecowton1 & @jamesgorrie, which addresses these concerns.

**Comment by bryophyta** [comment]
@mxdvl is this a PR that you want to merge into main before #7003? Or do you want to merge #7003 into this branch before merging this to main?

**Comment by mxdvl** [comment]
> @mxdvl is this a PR that you want to merge into main before #7003? Or are you intending to merge #7003 into this branch before merging this to main?

I’m thinking that this can be superceded by #7100

**Comment by bryophyta** [comment]
> I’m thinking that this can be superceded by https://github.com/guardian/dotcom-rendering/pull/7100

Okay thanks! In that case I'll add the `prnouncer-ignore` label to this PR for the time being, but obv feel free to remove it if you do want this PR to be reviewed!

**Pull request closed without merging**